lib/book.rb

```diff
@@ -0,0 +1,19 @@
+class Book
+  attr_reader :author,
+              :title,
+              :publication_date
+              
+  attr_accessor:checkout_count
+
+  def initialize(info)
+    @author = info[:author_first_name] + ' ' + info[:author_last_name]
+    @title = info[:title]
+    publication_date = if info[:publication_date].split(', ').length == 2
```

Comment:
Since the publication year will always be the last four characters of the publication date (see the spec), you could do something like this: `@publication_year = info[:publication_date][-4..-1]`